src/main/kotlin/it/pagopa/wallet/services/WalletService.kt

```diff
@@ -125,6 +133,156 @@ class WalletService(
             }
     }
 
+    fun validateWallet(
+        orderId: UUID,
+        walletId: UUID
+    ): Mono<Pair<WalletVerifyRequestsResponseDto, LoggedAction<Wallet>>> {
+        val correlationId = UUID.randomUUID()
+        return mono { npgSessionRedisTemplate.findById(orderId.toString()) }
+            .switchIfEmpty { Mono.error(SessionNotFoundException(orderId)) }
+            .flatMap { session ->
+                walletRepository
+                    .findById(walletId.toString())
+                    .switchIfEmpty { Mono.error(WalletNotFoundException(WalletId(walletId))) }
+                    .filter { session.walletId == it.id }
+                    .switchIfEmpty {
+                        Mono.error(
+                            WalletSessionMismatchException(session.sessionId, WalletId(walletId))
+                        )
+                    }
+                    .map { wallet -> wallet to session }
+                    .map { (wallet, session) -> Pair(wallet.toDomain(), session) }
+                    .filter { (wallet) -> wallet.status == WalletStatusDto.INITIALIZED }
+                    .switchIfEmpty { Mono.error(WalletConflictStatusException(WalletId(walletId))) }
+                    .flatMap { (wallet, session) ->
+                        ecommercePaymentMethodsClient
+                            .getPaymentMethodById(wallet.paymentMethodId.toString())
+                            .map { method -> Tuples.of(wallet, session, method) }
+                    }
+                    .flatMap { wallet_session_method ->
+                        val wallet = wallet_session_method.t1
+                        val session = wallet_session_method.t2
+                        val method = wallet_session_method.t3
+                        when (method.name) {
+                            "CARDS" ->
```

Comment:
@infantesimone is this the right way to check if a wallet is associated to a card payment method?